### PR TITLE
adds cargo capacity to cargo window for vehicles

### PR DESCRIPTION
**Pull request by LeftofZen**
Implements the feature requested in #1271.
I've added screenshots of my implementation in that issue.

### CODE REVIEW

File: src/OpenLoco/Vehicles/Vehicle.h [hunk 1]
```diff
@@ -312,6 +312,8 @@ namespace OpenLoco::Vehicles
         OrderRingView getCurrentOrders() const;
         bool isPlaced() const { return tile_x != -1 && !(var_38 & Flags38::isGhost); }
         char* generateCargoTotalString(char* buffer);
+        char* generateCargoCapacityString(char* buffer);
+        char* cargoLUTToString(uint32_t cargoTotals[32], char* buffer);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/Vehicle.h:316]: Can you switch to a std::array for this. I don't like passing c arrays around.

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 2]
```diff
@@ -3245,6 +3246,33 @@ namespace OpenLoco::Vehicles
...
+            if (body->primaryCargo.type != 0xFF)
+            {
+                cargoTotals[body->primaryCargo.type] += body->primaryCargo.maxQty;
+            }
+        }
+
+        return cargoLUTToString(cargoTotals, buffer);
+    }
+
+    char* VehicleHead::cargoLUTToString(std::array<uint32_t, 32>& cargoTotals, char* buffer)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:3274]: You've switched the declaration to use ObjectManager::getMaxObjects(ObjectType::cargo) but not the definition. Can you keep it consistent.

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/VehicleHead.cpp:3274]: sorry I missed this reference, do you mind if I alias the type out

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/VehicleHead.cpp:3274]: something like `using CargoTotalArray = std::array<uint32_t, ObjectManager::getMaxObjects(ObjectType::cargo)>;`

Comment (duncanspumpkin) [reply src/OpenLoco/Vehicles/VehicleHead.cpp]: Sure.

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/VehicleHead.cpp]: done

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**